Lib/test/test_configparser.py

```diff
@@ -663,8 +664,16 @@ def test_write(self):
 
         cf = self.fromstring(config_string)
         for space_around_delimiters in (True, False):
-            output = io.StringIO()
-            cf.write(output, space_around_delimiters=space_around_delimiters)
+            if testfile is None:
+                output = io.StringIO()
+            else:
+                output = open(testfile, 'w')
+            try:
+                cf.write(output,
+                         space_around_delimiters=space_around_delimiters)
```

Comment:
To test that cf.write can work given a filename, you must call with the filename.
The fix here would be to replace else to finally with
```
else:
    output = testfile
cf.write(output, ...
```
But see below.